server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -522,7 +519,7 @@ private boolean exists(final Set<String> keys, final SecureSettings secureSettin
      * @return true if the setting including fallback settings is present in the given settings instance, otherwise false
      */
     public boolean existsOrFallbackExists(final Settings settings) {
-        return settings.keySet().contains(getKey()) || (fallbackSetting != null && fallbackSetting.existsOrFallbackExists(settings));
```

Comment:
> Looks good to me, but can you please double check if this
>> Following the default implementation of exists, with this PR secure settings are always excluded
>
> is OK? In serverless too?

I've checked serverless, there's no usage of this beyond what's in the submodule. Also, this is a lot older.